apps/api/src/app/events/usecases/trigger-event/message-matcher.service.ts

```diff
@@ -112,6 +132,28 @@ export class MessageMatcher {
     return !!(await findAsync(asyncFilters, (i) => this.processFilter(variables, i, command)));
   }
 
+  private async processIsOnline(
+    filter: IRealtimeOnlineFilterPart | IOnlineInLastFilterPart,
+    command: SendMessageCommand
+  ): Promise<boolean> {
+    const subscriber = await this.subscriberRepository.findOne({
+      _id: command.subscriberId,
+      _organizationId: command.organizationId,
+      _environmentId: command.environmentId,
+    });
+
+    const isOnlineMatch = subscriber?.isOnline === filter.value;
+    if (filter.on === 'isOnline') {
+      return isOnlineMatch;
+    }
+
+    const currentDate = new Date();
+    const lastOnlineAt = subscriber?.lastOnlineAt ? parseISO(subscriber?.lastOnlineAt) : new Date();
+    const diff = differenceIn(currentDate, lastOnlineAt, filter.timeOperator);
+
+    return subscriber?.isOnline || (!subscriber?.isOnline && diff <= filter.value);
```

Comment:
for the `isOnlineInLast` we do check:
- if the subscriber is online
- or if the subscriber is offline but was online in the last X minutes, hours, days